c++/src/OrcHdfsFile.cc

```diff
@@ -66,22 +64,22 @@ namespace orc {
         options = config->GetOptions();
       }
       hdfs::IoService * io_service = hdfs::IoService::New();
-      //Wrapping fs into a shared pointer to guarantee deletion
-      std::shared_ptr<hdfs::FileSystem> fs(hdfs::FileSystem::New(io_service, "", options));
-      if (!fs) {
+      //Wrapping file_system into a unique pointer to guarantee deletion
+      file_system = std::unique_ptr<hdfs::FileSystem>(hdfs::FileSystem::New(io_service, "", options));
+      if (!file_system) {
```

Comment:
Yes, unique pointer actually supports this notation (thanks to [operator bool](http://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool))